3.2/root/usr/share/container-scripts/mongodb/README.md

```diff
@@ -88,3 +91,45 @@ the values stored in the variables and the actual passwords. Whenever a database
+
+For example to build customized MongoDB database image `my-mongodb-centos7` with configuration in `~/image-configuration/` run:
+
+```
+$ s2i build ~/image-configuration/ centos/mongodb-32-centos7 my-mongodb-centos7
+```
+
+The directory passed to `s2i build` should contain one or more of the following directories::
+- `mongodb-cfg/`
+  - when running `run-mongod` or `run-mongod-replication` commands contained `mongod.conf` file is used for `mongod` configuration
+    - `envsubst` command is run on this file to still allow customization of the image using environment variables
+    - custom configuration file does not affect name of replica set - it has to be set in `MONGODB_REPLICA_NAME` environment variable
+    - it is not possible to configure SSL using custom configuration file
+
+- `mongodb-pre-init/`
+  - contained shell scripts (`*.sh`) are sourced before `mongod` server is started
+
+- `mongodb-init/`
+  - contained shell scripts (`*.sh`) are sourced when `mongod` server is started
+    - `run-mongod` command doesn't have enabled authentication in this phase
+    - `run-mongod-replication` command has enabled authentication in this phase
+  - these scripts are skipped if `run-mongod-replication` is run with already initialized data directory
+
+Variables that can be used in the scripts provided to s2i:
+
+- `mongo_common_args` -- contains arguments for the `mongod` server (changing this can break existing customization scripts, e.g. default scripts)
+- `$MEMBER_ID` -- contains 'id' of the container. It is defined only in scripts for replication (`run-mongod-replication` command) and has different value for each container in a replicaset cluster. Customization scripts are run by all containers in replicaset - `MEMBER_ID` can be used to write scripts which are run only by some container.
+
+During `s2i build` all provided files are copied into `/opt/app-root/src` directory in the new image. If some configuration files are present in destination directory, files with the same name are overwritten. Also only one file with the same name can be used for customization and user provided files are preferred over default files in `/usr/share/container-scripts/mongodb/`- so it is possible to overwrite them.
+
+Same configuration directory structure can be used to customize the image every time the image is started using `docker run`. The directory have to be mounted into `/opt/app-root/src/` in the image (`-v ./image-configuration/:/opt/app-root/src/`). This overwrites customization built into the image.
+
+Mounting custom configuration file
+---------------------------------
+
+It is allowed to use custom configuration file for `mongod` server. To use it in container it has to be mounted into `/etc/mongod.conf`. For example to use configuration file stored in `/home/user` directory use this option for `docker run` command: `-v /home/user/mongod.conf:/etc/mongod.conf:Z`.
```

Comment:
@hhorak this is either wrong, or isn't what you described re: configmap mounting (and won't work for mounting configmaps since as discussed, we can't mount a single file, we need to mount the entire /etc dir)